lib/logthrdest/logthrdestdrv.c

```diff
@@ -1061,7 +1061,7 @@ log_threaded_dest_driver_init_method(LogPipe *s)
 
   self->under_termination = FALSE;
 
-  if (cfg && self->time_reopen == -1)
+  if (self->time_reopen == -1)
```

Comment:
Do not remove that check. You may know that is the case, still the compiler does not.

The logic is the following the `cfg` might be `NULL` or not. I have no clue, nor can I find out in compile time so I assume both. The next step is, well you dereference a pointer that could be `NULL`, that is undefined behavior. At that point compiler are free to do what its heart desires, and they do if they smell opportunity of performance gain. (likely this does what you would expect, which is worse because let you think it is okay).

Reply:
This is new to me.

I am just thinking, why isn't this a problem in every one of our variables in every one of our functions? Like the `self` here, or we can find a lot of examples. We do not check for every variable to not be `NULL` in the beginning of our functions.

Why is this case different, and how can someone detect it?

Reply:
We discussed this IRL, and added an assert to the getter.

Reply:
btw, I think the compiler behavior is the other way around, e.g. if it does see an unconditional dereference, it would assume the pointer is NOT null and thus drop the NULL checks, as explained here: https://lwn.net/Articles/342330/

I don't see how the code with the original change would cause suprising code to be generated.